File: LeetCode75/hashmap and sets/uniqueNumberOfOccurences.js
// Given an array of integers arr, return true if the number of occurrences of each value in the array is unique or false otherwise.

/**
 * @param {number[]} arr
 * @return {boolean}
 */
function uniqueOccurrences(arr) {
  // Create a hash map to store element counts
  const valueCounts = {};

  // Count occurrences of each element
  for (const val of arr) {
    valueCounts[val] = (valueCounts[val] || 0) + 1;
  }

  // Convert hash map values to an array
  const countsList = Object.values(valueCounts);

  // Check if the number of unique elements (set size) is equal to the number of occurrences (list length)
  return new Set(countsList).size === countsList.length;
}

File: LeetCode75/hashmap and sets/uniqueNumberOfOccurences.py
class Solution(object):
    def uniqueOccurrences(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """
        valsCount = {}
        for val in arr:
            valsCount[val] = 1 + valsCount.get(val, 0)
        
        valuesList = list(valsCount.values())

        return len(valuesList) == len(set(valuesList))


        